annif/project.py

```diff
@@ -120,8 +123,14 @@ def _suggest_with_backend(self, text, backend_params):
 
     @property
     def analyzer(self):
-        if self._analyzer is None and self.analyzer_spec:
-            self._analyzer = annif.analyzer.get_analyzer(self.analyzer_spec)
+        if self._analyzer is None:
+            if self.analyzer_spec:
+                self._analyzer = annif.analyzer.get_analyzer(
+                    self.analyzer_spec)
+            elif self.backend.needs_subject_vectorizer:
```

Comment:
I don't think this is the right condition to test, maybe not even the right place for it. It happens to work for the case where an analyzer is needed within the AnnifProject class, but there are other situations in backend-specific code where an analyzer is needed as well, irrespective of `needs_subject_vectorizer`:

* [fasttext.py](https://github.com/NatLibFi/Annif/blob/master/annif/backend/fasttext.py#L77)
* [mixins.py](https://github.com/NatLibFi/Annif/blob/master/annif/backend/mixins.py#L21)
* [vw_multi.py](https://github.com/NatLibFi/Annif/blob/master/annif/backend/vw_multi.py#L57)

Alternatively, backends could have a `needs_analyzer` field that indicates that they are making use of the analyzer, then the `if` clause above could check for that.